### PR TITLE
Felinids don't like getting sprayed with water, take 2

### PR DESCRIPTION
**Pull request by Ryll-Ryll**
## About The Pull Request
This PR is an ode to @KathyRyals and their PR #59506, which was itself originally an ode to me, but who's keeping track of that? This is a much leaner implementation of that PR that simply gives felinids who are squirted with water a spray bottle (read: exposed to water as a vapor) a 1 decisecond Stun (as in the Stun status effect, so no knockdown or anything), so that you can interrupt their do_afters (if they're, for example, climbing on a table, as cats tend to do). It also gives them a mild negative moodlet.

This species change is approved by @Mothblocks 

## Why It's Good For The Game
Very mild little bit of silliness and species humor

## Changelog

:cl: Cimika and Ryll/Shaps
add: You can now spray felinids with water from a squirt bottle to interrupt any actions they're in the middle of and give them a mild negative moodlet. No climbing on the counter!
/:cl:

### CODE REVIEW

**Comment by ATH1909** [comment]
does this apply to extinguisher sprays, or are those touch-based?

**Comment by Acer202** [comment]
Right now this is a spacesuit-penetrating, block chance circumventing, instant item drop water spray. Cat SEC: Destroyed. Cat traitors: Destroyed. Please reconsider, Shaps! This is crazy powerful against felinids.

**Comment by Ryll-Ryll** [comment]
Didn't realize Stun also included blocking hands, turns out it wasn't as simple as I expected! Will look into.

**Comment by ATH1909** [comment]
> Right now this is a spacesuit-penetrating, block chance circumventing, instant item drop water spray. Cat SEC: Destroyed. Cat traitors: Destroyed. Please reconsider, Shaps! This is crazy powerful against felinids.

I was actually fine with this until I remembered that being stunned makes you drop your items. Uh, would a 0.1 second immobilize and the sending of a signal or something that disrupts do_after()s work?

**Comment by KathyRyals** [comment]
Yes this is the reason why I had to make a whole new status effect in the original PR.

**Comment by necromanceranne** [comment]
Every time a stun chem emerges is another demonstration of how truly awful spray code is and how little actually defends against it when it is actually working properly.

**Comment by Wollywoger** [comment]
reading the title of this pr but then seeing the "interrupts any progress bar action" feature immediately made me go from actually wanting this in for the funny interactions, to me praying with every fiber of my body to not have this in
just cut the stun/progress bar deletion because dear god its the most low-tier (but geniunely frustrating) grief imaginable

imagine, like the catsec case, a tider interrupting a sec arrest by them using a spray bottle that immediately stops the seccie from cuffing and (possibly) letting the arrestee go free because the officer chose the funny anime race
at least with something like the laser pointer its mild enough that it'll spawn comedic interactions without giving the opportunity to significantly mess with a round/grief someone

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 1]
```diff
@@ -212,6 +212,9 @@
 	. = ..()
 	if(methods & TOUCH)
 		exposed_mob.extinguish_mob() // extinguish removes all fire stacks
+	if((methods & VAPOR) && isfelinid(exposed_mob))
```
Comment (Wollywoger) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm:215]: I'm not very well versed in /tg/ code, but from what i've heard from a friend "VAPOR" is also used for smoke machines and possibly water vapor, which could result in a stunlock

Comment (CRITAWAKETS) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm:215]: actually its foam
which is even stronger for things like grenades but cant get smoke machined as easily
so less grief, more powergame

**Comment by Acer202** [comment]
> also hopefully the comments above are incorrect with stuns of this type causing you to drop items, as this would make it actually abuseable

This PR is meant to be a lighthearted, funny thing, but the reality is just more people griefing felinids and I'm really hoping it's not added as well. I booted up a private server to test the code to verify the item-dropping, and sure enough, you can be wearing a syndie hardsuit wielding a desword and one spritz of water is enough to make you drop everything instantly. I didn't even consider the handcuffing catSEC problem, but that makes me want it less.

**Comment by Farquaar** [comment]
> 
> 
> the reality is just more people griefing felinids

Intended feature :^)

**Comment by Ryll-Ryll** [comment]
> > also hopefully the comments above are incorrect with stuns of this type causing you to drop items, as this would make it actually abuseable
> 
> This PR is meant to be a lighthearted, funny thing, but the reality is just more people griefing felinids and I'm really hoping it's not added as well. I booted up a private server to test the code to verify the item-dropping, and sure enough, you can be wearing a syndie hardsuit wielding a desword and one spritz of water is enough to make you drop everything instantly. I didn't even consider the handcuffing catSEC problem, but that makes me want it less.

Will you chill out? I already directly said the item dropping is not intentional and that I'd remove it.

May I remind people that interrupting do_after's can also just as easily be accomplished literally just by walking into someone

**Comment by bluetshirtguy** [comment]
Can you add the lizard pr now?

File: code/datums/status_effects/debuffs.dm [hunk 2]
```diff
@@ -91,6 +91,19 @@
 	REMOVE_TRAIT(owner, TRAIT_HANDS_BLOCKED, TRAIT_STATUS_EFFECT(id))
 	return ..()
 
+/// Startled - only applies TRAIT_INCAPACITATED, such that it interrupts/blocks do_afters, but doesn't make you drop items or freeze you in place like Stun
```
Comment (ATH1909) [inline code/datums/status_effects/debuffs.dm:94]: you, uh, forgot to actually use this new condition, the spray still stuns you

**Comment by Wollywoger** [comment]
with the new change of adding the solely interrupting do-afters i'd say this pr is a fair bit better imo
if felinids ever get the ability to hiss, it would be perfect if felinids had a chance to hiss upon being sprayed with water

**Comment by tralezab** [comment]
you should put this in the felinid.ts for their species

**Pull request closed without merging**